src/components/Dashboard/UpcomingBookings.vue

```diff
@@ -87,11 +90,11 @@ onMounted(() => {
         <button class="button red-bg" @click="router.push('/book/realtime')"> Book Slot </button>
     </div>
     <div class="observations upcoming-obs">
-        <h3><span v-if="!requestGroups.length">No </span>Pending Requests</h3>
+        <h3><span v-if="Object.keys(pendingScheduledObservations).length == 0">No </span>Pending Requests</h3>
         <div class="table-summary">
-            <div v-for="requestGroup in requestGroups" :key="requestGroup.id">
-              <div v-for="request in requestGroup.requests" :key="request.id">
-                <div>{{ request.configurations[0].target.name }}</div>
+            <div v-for="scheduledObs in pendingScheduledObservations" :key="scheduledObs.id">
+              <div v-for="request in scheduledObs.request.configurations" :key="request.id">
```

Comment:
if you are looping over configurations, then you should call the variable "configuration", not "request". The configuration has a target, not the request.